cpp/src/binaryop/binaryop.cpp

```diff
@@ -323,7 +323,7 @@ std::unique_ptr<column> binary_operation(scalar const& lhs,
                                          rmm::mr::device_memory_resource* mr)
 {
   return binops::compiled::binary_operation<scalar, column_view>(
-    lhs, rhs, op, output_type, cudf::default_stream_value, mr);
+    lhs, rhs, op, output_type, stream, mr);
```

Comment:
These functions were incorrectly using the default stream instead of the stream passed into the function.